src/main/java/core/basesyntax/strategy/OperationStrategy.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> map;
+
+    public OperationStrategy(OperationHandler... handlers) {
+        this.map = new HashMap<>();
+        for (OperationHandler handler : handlers) {
+            map.put(handler.getType(), handler);
+        }
```

Comment:
if you want you can do it like this:
        map = Arrays.stream(handlers)
                .collect(Collectors.toMap(OperationHandler::getType, Function.identity()));